File: JavaDemoGithubPractice/app/src/main/java/com/example/javademogithubpractice/mvp/contract/ISplashContract.java


package com.example.javademogithubpractice.mvp.contract;


public interface ISplashContract {

    interface View extends IBaseContract.View{
        void showMainPage();
    }

    interface Presenter extends IBaseContract.Presenter<ISplashContract.View>{
        void getUser();
        void saveAccessToken(String accessToken, String scope, int expireIn);
    }

}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/mvp/contract/ISplashContract.kt
package com.example.kotlindemogithubproject.mvp.contract


interface ISplashContract {

    interface View : IBaseContract.View {
        fun showMainPage()
    }

    interface Presenter : IBaseContract.Presenter<View> {

        fun getUser()

        fun saveAccessToken(accessToken: String, scope: String, expireIn: Int)

    }

}
